backend/src/main/kotlin/dev/kviklet/kviklet/service/JDBCExecutor.kt

```diff
@@ -238,10 +242,26 @@ class JDBCExecutor {
 
             rdsUtilities = RdsUtilities.builder()
                 .region(region)
-                .credentialsProvider(DefaultCredentialsProvider.create())
+                .credentialsProvider(createCredentialsProvider())
                 .build()
         }
 
+        private fun createCredentialsProvider(): AwsCredentialsProvider = if (roleArn != null) {
```

Comment:
I feel like this should be a null or empty check because if you edit the connection this might be set to an empty string.

Reply:
good point, will fix that